docs: changelog 2024-12-20 (#2660)
neondatabase/website: content/changelog/2024-12-20.md

If you use the [Neon CLI](https://neon.tech/docs/reference/neon-cli), here’s a friendly reminder to upgrade to the latest version. The current version is 2.6.0, and all users should be on a 2.x version. You can check your Neon CLI version with the following command:

```bash
neon --version
2.6.0
```

For upgrade instructions, see [Neon CLI upgrade](https://neon.tech/docs/reference/cli-install#upgrade).

If you use the Neon CLI in CI/CD tools like GitHub Actions, you can safely pin it to `latest`, as we prioritize stability for CI/CD processes. For example, in a GitHub Actions workflow, you can install the latest version with `npm`:

```yaml
- name: Install Neon CLI
  run: npm install -g neonctl@latest
```

<details>

<summary>**Fixes & improvements**</summary>

- **Drizzle Studio update**

  The Drizzle Studio integration that powers the **Tables** page in the Neon Console has been updated. For the latest improvements and fixes, see the [Neon Drizzle Studio Integration Changelog](https://github.com/neondatabase/neon-drizzle-studio-changelog/blob/main/CHANGELOG.md).

- **Console updates**

  - Added a link to the Neon API Reference in the information resources menu of the console.

>   - The Support ticket modal now supports file attachments in the Neon Console (attachments must not exceed 5 MB).

Reviewer: ```suggestion
  - The Support ticket modal now supports file attachments in the Neon Console. Attachments must not exceed 5 MB.
```

>   - Sorting is now available for more columns in the branches table on the **Branches** page.

Reviewer: ```suggestion
  - You can now sort columns in the branches table on the **Branches** page.
```

  Support is now available for transferring projects from Neon personal accounts to Azure-created Neon organizations. Project transfers are only supported for Neon projects created in an Azure region. For more details, see [Transfer Neon projects to an Azure-created Neon organization](/docs/introduction/billing-azure-marketplace#transfer-existing-neon-projects-to-an-azure-created-neon-organization).

- **Neon API**

  We added `204` response definitions for `DELETE` endpoints for scenarios where a branch, database, role, or endpoint does not exist or has already been deleted. A 204 response code indicates a successful operation that does not return content.

- **Fixes**

  - Fixed an issue in the console where an error toast was not displayed when offline or when the server was unreachable. A `Network error` is now returned in such cases.
  - Corrected a usability issue where the **Project Creation** modal defaulted to an error state, requiring users to adjust autoscaling settings before creating a project.
  - Resolved an issue where users encountered a confusing error message when attempting to link a Microsoft account already logged in with the same email.

>   - Fixed a drag-to-zoom failure on **Monitoring** page graphs that occurred when dragging from left to right and releasing the cursor outside of the graph area.

Reviewer: ```suggestion
  - Fixed an issue with drag-to-zoom on **Monitoring** page graphs. The issue occurred when dragging from left to right and releasing the cursor outside of the graph area.
```
